code/modules/clothing/spacesuits/rig/modules/utility.dm

```diff
@@ -740,6 +740,49 @@
 		H.forceMove(leapEnd)
 		return TRUE
 
+/obj/item/rig_module/springstep
+	name = "springstep module"
+	desc = "A mix of electric stimulators and metallic actuators, this module detects a footfall and sends a signal to the muscle to contract to increase baseline moving speed."
+	icon_state = "actuators"
+
+	interface_name = "springstep module"
+	interface_desc = "Allows you to move faster than normal."
+
+	construction_cost = list(DEFAULT_WALL_MATERIAL=15000, MATERIAL_GLASS = 1250)
+	construction_time = 300
+
+	disruptive = FALSE
+
+	use_power_cost = 0
+	module_cooldown = 5
+	var/step_power_cost = 15
```

Comment:
Needs to be far more expensive.

Additionally, there's a pretty big problem: this doesn't check for a maximum move delay. This means that this module would be absolutely stupid on an m'sai tajara, and even more stupid on caffeine.

Reply:
increased movement cost to 25 per step, put in a maximum move delay that's a bit slower than m'sai running speed